web/docs/understand/operators/sinks/tui.md

```diff
@@ -0,0 +1,60 @@
+# tui
+
+Renders pipeline data in a terminal user interface (TUI).
+
+## Synopsis
+
+```
+tui [-f|--fullscreen=<bool>] [-h|--height=<int>] [-w|--width=<int>]
+```
+
+## Description
+
+The `tui` operator renders the data in the pipeline in a terminal user interface
+(TUI).
+
+Use the arrow keys or `j`/`k`/`h`/`l` to navigate through the cells. You can
+also use the mouse to scroll vertically and horizontally, but moving the pointer
+relative off the center of the TUI in a specific direction. For example, moving
+the point to the right from the center point scrolls to the right.
+
+Press `?` to get an interactive help.
+
+The implementation uses the [FTXUI](https://arthursonzogni.github.io/FTXUI/)
+library for functional terminal interfaces in C++.
+
+### `-f|--fullscreen`
+
+Renders the TUI in fullscreen mode, as opposed to fitting it into the terminal.
+
+### `-h|--height=<int>`
+
+The height of the TUI. Only actionable when `-w|--width` is also present.
+
+Defaults to 0, which means automatic detection.
+
+### `-w|--width=<int>`
+
+The width of the TUI. Only actionable when `-h|--height` is also present.
+
+Defaults to 0, which means automatic detection.
+
+## Example
+
+Show the pipeline results in a TUI:
+
+```bash
+tui
+```
+
+Show the pipeline results in a TUI, in fullscreen mode:
+
+```bash
+tui -f true
```

Comment:
It would be nice to just have `tui -f` here.

@dominiklohmann I looked at `option_set_parser` and was wondering: is there something that fundamentally prevents boolean options with having to provide the `true` value? Or was this just skipped initially due to the higher complexity?

Reply:
We want to get rid of letting operators parse the pipeline themselves entirely in favor of letting operators return a description of the arguments they support from which a parser can be generated. All the things we currently have for parsing are driven by demand—and so far, nobody tried parsing flags. Feel free to improve the parser to suit your needs, or to just parse the `-f` option outside of the option set parser.